sdk/core/azure-core-test/src/test/java/com/azure/core/test/implementation/RestProxyWithMockTests.java

```diff
@@ -219,7 +221,7 @@ public void serviceErrorWithResponseContentTypeBadJSON() {
             fail();
         } catch (HttpResponseException ex) {
             assertContains(ex.getMessage(), "Status code 200");
-            assertContains(ex.getMessage(), "\"BAD JSON\"");
+            assertContains(ex.getMessage(), "BAD JSON");
```

Comment:
Where does 'BAD JSON' come from?

Reply:
The rest proxy test mock some service response. This is coming from the mock service call. 
```
        ServiceErrorWithCharsetService service = RestProxy.create(
                ServiceErrorWithCharsetService.class,
                new HttpPipelineBuilder().httpClient(new SimpleMockHttpClient() {
                    @Override
                    public Mono<HttpResponse> send(HttpRequest request) {
                        HttpHeaders headers = new HttpHeaders().put("Content-Type", "application/json");

                        HttpResponse response = new MockHttpResponse(request, 200, headers, "BAD JSON".getBytes(StandardCharsets.UTF_8));
                        return Mono.just(response);
                    }
                }).build());
```